compiler/rustc_errors/src/lib.rs

```diff
@@ -768,13 +768,7 @@ impl DiagCtxt {
                     format!("invalid level in `stash_diagnostic`: {:?}", diag.level),
                 );
             }
-            Error => {
-                // This `unchecked_error_guaranteed` is valid. It is where the
-                // `ErrorGuaranteed` for stashed errors originates. See
-                // `DiagCtxtInner::drop`.
-                #[allow(deprecated)]
-                Some(ErrorGuaranteed::unchecked_error_guaranteed())
-            }
+            Error => Some(self.span_delayed_bug(span, "stashing {key:?}")),
```

Comment:
Stashed diagnostics, argh, such a PITA.

Having to create a delayed bug for every single one is unfortunate. Is it possible to only do that when `-Ztreat-err-as-bug` is enabled? You haven't described the debugging much so I'm only guessing at what you're doing. Also a brief comment here explaining why a delayed bug is preferable to an unchecked `ErrorGuarantee` would be useful.

Reply:
I actually like the use of `span_delayed_bug` over `unchecked_error_guaranteed`, and it's not like it's a perf issue